architecture/build.gradle

```diff
@@ -22,6 +22,10 @@ android {
             consumerProguardFiles 'proguard-rules.pro'
         }
     }
+    compileOptions {
```

Comment:
Will this potentially break anything else? cc: @colintheshots

Reply:
No, this will simply enable desugaring to support Java 8 language features, which we should've been using anyway. We may also want to set the Kotlin JVM target for both modules. `kotlinOptions.jvmTarget = '1.8'` This goes under the `android` level.